test/vinyl-luatest/gh_6568_replica_initial_join_deletion_of_compacted_run_files_test.lua

```diff
@@ -0,0 +1,85 @@
+local cluster = require('test.luatest_helpers.cluster')
+local fio = require('fio')
+local t = require('luatest')
+
+local g = t.group('gh-6568-replica-initial-join-deletion-of-compacted-run-files')
+
+local s_id = 0
+g.before_all(function()
+    local helpers = require('test.luatest_helpers')
+
+    g.cluster = cluster:new({})
+
+    local master_uri = helpers.instance_uri('master')
+    local master_box_cfg = {
+        checkpoint_count = 1,
+        listen           = master_uri,
+    }
+    g.master = g.cluster:build_and_add_server({alias   = 'master',
+                                               box_cfg = master_box_cfg})
+
+    local replica_box_cfg = {
+        listen       = helpers.instance_uri('replica'),
+        vinyl_memory = 128,
+        replication  = {
+            master_uri,
+        },
+        read_only    = true,
+    }
+    g.replica = g.cluster:build_and_add_server({alias   = 'replica',
+                                                box_cfg = replica_box_cfg})
+
+    g.master:start()
+    s_id = g.master:exec(function()
+        local s = box.schema.space.create('test', {engine = 'vinyl'})
+        s:format({{name = '1', type = 'number'},
+                  {name = '2', type = 'number'},
+                  {name = '3', type = 'number'},
+                  {name = '4', type = 'number'},
+                  {name = '5', type = 'number'},
+                  {name = '6', type = 'number'},
+                  {name = '7', type = 'number'},
+                  {name = '8', type = 'number'}
+                 })
+        s:create_index('primary')
+
+        for i = 1, 64 do s:insert{i, 1e15, 1e15, 1e15, 1e15, 1e15, 1e15, 1e15} end
```

Comment:
You could use a string for padding:

```lua
for i = 1, 64 do s:insert{i, string.rep('x', 100)} end
```

Reply:
Fixed: that's a really elegant approach, did not think about that – thank you for sharing.